packages/core/src/interpreter.ts

```diff
@@ -357,8 +309,8 @@ export class Interpreter<
     if (typeof nextListenerOrObserver === 'function') {
       listener = nextListenerOrObserver;
     } else {
-      listener = nextListenerOrObserver.next.bind(nextListenerOrObserver);
-      resolvedCompleteListener = nextListenerOrObserver.complete.bind(
+      listener = nextListenerOrObserver.next!.bind(nextListenerOrObserver);
```

Comment:
Observer type has [`next` defined as optional](https://github.com/davidkpiano/xstate/blob/13918f4b3a49d5917fa1b682615d62f63fc57808/packages/core/src/types.ts#L1066), either it should become required or this line here should use `?` rather than `!`